### DISCUSSION TITLE
'cloze'-option

### DISCUSSION THREAD

**Discussion by aleksejrs** [mortii/anki-morphs#12] [Ideas]
Or maybe track active recall and passive recognition in parallel.

**Comment by mortii** [comment]
Could you provide an example?

I added the [statistics page](https://mortii.github.io/anki-morphs/user_guide/usage/statistics.html) back to the guide, are there any specific changes you would like to see on the statistics pop-up?

**Comment by aleksejrs** [comment]
For example, I have this card:

```
What is a tautline hitch useful for?

Tying to {{c1::a post or ring}} in an adjustable way.
```

When it becomes a mature card, I might still not be able to name the knot ("tautline hitch") or produce the phrase "in an adjustable way", so those words should not be considered mature morphs just because that card containing them in the question is mature. Maybe they could still be considered somewhat known though, so that cards with the same words are considered a little easier because of that.

However, those phrases add length to the card, and it is useful to know them before starting to learn that card, so when the card is new, they should be considered unknown and lowering the priority of the card, but not be used as focus morphs.

Also, when the card is unmature (and *maybe* when it is mature and has been reviewed today), new cards that have any of those non-mature morphs in the answer should not be introduced.


A [movies2anki](https://github.com/kelciour/movies2anki) card doesn't teach much about its morphs, so they shouldn't become mature, either.

**Comment by mortii** [comment]
That is an interesting problem. So you essentially want to use cards that are not specific to language learning, i.e. you don't care too much about knowing the individual words themselves, but rather the concept the words help describe?

My first impression is that since the use-case of the card is different from language learning it probably should not be used in a language learning context, and the card should just be ignored by AnkiMorphs completely because it might be misleading (as you described).

I could see this being an extremely common problem and it should definitely be addressed in the FAQ.

**Comment by aleksejrs** [comment]
I hope my comment won't look rude or confusing. It seems like you have a certain card format and card creation process in mind when developing the add-on, and it might be targeting only vocabulary building?

> So you essentially want to use cards that are not specific to language learning, i.e. you don't care too much about knowing the individual words themselves, but rather the concept the words help describe?

You can learn words and concepts at the same time, and the add-on can still help.

I've [made my MorphMan use also two-word morphs](https://github.com/mortii/anki-morphs/discussions/11) and count those that are likely to be in an easier language as shorter and easier than those that are likely to be in a harder language. Cards that don't benefit from that don't seem like a very good idea for language learning to me, at least with an analytical language like English, where a word can have tens of context-dependent meanings, but the word order is stable enough to count and learn word combinations for fluency.

There is no 1:1 correspondence between languages, and I guess not even between the meaning of most words to two people in the same language, so learning a language involves learning new concepts through context and explanation.

https://en.wikipedia.org/wiki/Vocabulary#Productive_and_receptive_knowledge
Cards can teach translation, pronunciation or transcription, which are still language stuff, and the ordinary MorphMan wouldn't know which one to make first, but my edit takes tags, note types and languages into account. Of course, there are also decks, but I've only just started using them again because of FSRS, and not all cards are equally important, and MorphMan had been helping me with the less urgent cards rather than just finding the few "k+1" cards that I guess *might* come from a course somebody else had made for you, but are really hard to make out of examples from random dictionary articles.

> My first impression is that since the use-case of the card is different from language learning it probably should not be used in a language learning context, and the card should just be ignored by AnkiMorphs completely because it might be misleading (as you described).

If I had only ten new cards in total, or made cards only from an ordered course, I could order them without complicated add-ons.

Let's try this example:

Q: [phrase in second language]
A: [phrase in third language]

I may not know some words in the question well, and learning this card will **not** help me produce sentences with them. Cards for the language used in the question will be separate cloze deletion cards.

But that doesn't mean the add-on should just ignore the card or the question because I am likely to understand the words in the question. Shorter cards should be introduced before the longer ones, because they take less time and often contain the basics, and the easier I understand the question, the easier the card will be to learn.

**Comment by mortii** [comment]
This is a great discussion, thanks you so much!

Could one potential solution be to have a "only read content of cloze brackets"-option? This would be the almost the inverse of the existing "ignore brackets"-option.


So with your example:
> What is a tautline hitch useful for?
> 
> Tying to {{c1::a post or ring}} in an adjustable way.

Only "a post or ring" would be read by AnkiMorphs and the rest be ignored?

That seems like a reasonable feature. There could be a checkbox with the title "Cloze" or something in the note-filter option to enable/disable it.

**Comment by aleksejrs** [comment]
That minimal solution wouldn't
* penalize cards that have a lot of context and thus take more time. Though maybe the morphemizer could return the number of words and characters in the note? I thought of making my morphemizer return both only the morphs in the brackets and all morphs as different lists, but that may be too hard to me (though I think it returns multiple copies of the list…).
* penalize cards with unknown morphs, preferring cards with known morphs.
* penalize cards with unmature morphs outside of the cloze brackets, although I think that is not really implemented in MorphMan except for focus morphs.

**Comment by mortii** [comment]
Ah, I just re-read your previous comment:


> When it becomes a mature card, I might still not be able to name the knot ("tautline hitch") or produce the phrase "in an adjustable way", so those words should not be considered mature morphs just because that card containing them in the question is mature. Maybe they could still be considered somewhat known though, so that cards with the same words are considered a little easier because of that.
> 
> However, those phrases add length to the card, and it is useful to know them before starting to learn that card, so when the card is new, they should be considered unknown and lowering the priority of the card, but not be used as focus morphs.
> 
> Also, when the card is unmature (and _maybe_ when it is mature and has been reviewed today), new cards that have any of those non-mature morphs in the answer should not be introduced.
> 
> A [movies2anki](https://github.com/kelciour/movies2anki) card doesn't teach much about its morphs, so they shouldn't become mature, either.

So you want to read all the text and sort the card accordingly, but only mark the words in the cloze brackets as known? This also seems doable.

**Reply by aleksejrs** [reply]
Right. And the cloze hint is also not part of the answer.

**Reply by mortii** [reply]
I'll add it to the roadmap

**Comment by mortii** [comment]
@aleksejrs I now think that this feature would be overfitting and too detrimental to the overall algorithm unfortunately, so I won't be implementing this, sorry :slightly_frowning_face:
